src/mqt/bench/utils.py

```diff
@@ -313,7 +313,7 @@ def calc_qubit_index(qargs: list[Qubit], qregs: list[QuantumRegister], index: in
 def calc_supermarq_features(
     qc: QuantumCircuit,
 ) -> SupermarqFeatures:
-    qc.remove_final_measurements(inplace=True)
+    qc = qc.remove_final_measurements(inplace=False)
```

Comment:
Instead of removing barriers and measurements here (and now creating a copy of the circuit), you could just add an `if` statement in the loop over `qc.data` (L323) that skips over measurements and barriers. That would not alter the circuit and would not require a copy.
(You have to change one of the `_` to an actual loop index to get the name of the instruction)

Reply:
True, thanks for pointing that out. I modified the code accordingly. For that, I had to make minor modifications for the number of gates and depths calculations.